types/src/v0/impls/instance_state.rs

```diff
@@ -264,6 +267,25 @@ impl NodeState {
     }
 }
 
+/// NewType to hold upgrades and some convenience behavior.
+pub struct UpgradeMap(pub BTreeMap<Version, Upgrade>);
```

Comment:
IIUC this is intended only as test code due to the `unwrap`? Can we feature gate it as such so we don't at some point end up using it in normal code?